rubin_scheduler/scheduler/surveys/too_scripted_surveys.py

```diff
@@ -0,0 +1,691 @@
+__all__ = ("ToOScriptedSurvey", "gen_too_surveys")
+
+from copy import copy
+
+import numpy as np
+
+from rubin_scheduler.scheduler.surveys import BaseMarkovSurvey, ScriptedSurvey
+from rubin_scheduler.scheduler.utils import (
+    comcam_tessellate,
+    order_observations,
+    scheduled_observation,
+    thetaphi2xyz,
+    xyz2thetaphi,
+)
+from rubin_scheduler.site_models import _read_fields
+from rubin_scheduler.utils import _approx_ra_dec2_alt_az, _ra_dec2_hpid
+
+
+def rotx(theta, x, y, z):
+    """rotate the x,y,z points theta radians about x axis"""
+    sin_t = np.sin(theta)
+    cos_t = np.cos(theta)
+    xp = x
+    yp = y * cos_t + z * sin_t
+    zp = -y * sin_t + z * cos_t
+    return xp, yp, zp
+
+
+class ToOScriptedSurvey(ScriptedSurvey, BaseMarkovSurvey):
+    """If there is a new ToO event, generate a set of scripted
+    observations to try and follow it up.
+
+    Parameters
+    ----------
+    times : list of floats
+        The times after the detection that observations
+        should be attempted (hours)
+
+    alt_min : float
+        Do not attempt observations below this limit (degrees).
+        Note the telescope alt limit is 20 degrees, however, slew
+        and filter change time means alt_min here should be set higher
+        (otherwise, target will pass altitude check, but then fail
+        to observe by the time the telescope gets there).
+
+    """
+
+    def __init__(
+        self,
+        basis_functions,
+        followup_footprint=None,
+        nside=32,
+        reward_val=1e6,
+        times=[1, 2, 4, 24, 48],
+        filters_at_times=["gz", "gz", "gz", "gz", "gz", "gz"],
+        nvis=[1, 1, 1, 1, 6, 6],
+        exptimes=[30.0, 30, 30, 30, 30, 30],
```

Comment:
Also, it's kind of hard to understand how exptimes, nvis, and filters_at_times map together .. 
it seems like these set up 1 visit in each of g and z that are 30s each, at time 1 (then similar up to time 24). 
More documentation on this would be helpful.

Reply:
OK, I set a global variable `DEFAULT_EXP_TIME` so any ToO follow-ups that want to try and match the standard visit can easily. We'll probably want to further refine where ToO code goes since we'll want more active community input on it, and it is fairly rubin-specific. (So I imagine a lot of this file could go to another repo at some point.)